simulaqron/toolbox/manage_nodes.py

```diff
@@ -2,194 +2,232 @@
 import json
 from contextlib import closing
 from socket import AF_INET, SOCK_STREAM, socket
-from cqc.settings import get_config, set_config
 
 from simulaqron.toolbox import get_simulaqron_path
-from simulaqron.general.hostConfig import load_node_names, networkConfig
-from simulaqron.settings import Settings
+from simulaqron.settings import simulaqron_settings
 
 
-config_files = [Settings.CONF_APP_FILE, Settings.CONF_CQC_FILE, Settings.CONF_VNODE_FILE]
-node_config_file = Settings.CONF_NODES_FILE
+config_files = [simulaqron_settings.app_file, simulaqron_settings.cqc_file, simulaqron_settings.vnode_file]
+node_config_file = simulaqron_settings.nodes_file
 simulaqron_path = get_simulaqron_path.main()
 config_folder = "config"
 default_topology_file = os.path.join(simulaqron_path, config_folder, "topology.json")
 
 
-def add_node(name, hostname=None, app_port=None, cqc_port=None, vnode_port=None, neighbors=None):
-    """
-    Adds a node with the given name from the config files.
-    If the port numbers a not specified, unused ones will be chosen between 8000 and 9000.
-
-    :param name: str
-        Name of the node, e.g. Alice
-    :param hostname: str or None
-        Hostname, e.g. localhost (default) or 192.168.0.1
-    :param app_port: int or None
-        Port number for the application
-    :param cqc_port: int or None
-        Port number for the cqc server
-    :param vnode_port: int or None
-        Port number for the virtual node
-    :param neighbors: (list of str) or None
-        A list of neighbors, of this node, if None all current nodes in the network will be adjacent to the added node.
-    :return: None
-    """
-    if name in get_nodes():
-        raise ValueError("Cannot add node {}, already in the network.".format(name))
-
-    if hostname is None:
-        hostname = "localhost"
-
-    ports = [app_port, cqc_port, vnode_port]
-    for config_file, port in zip(config_files, ports):
-        if port is None:
-            port = _get_unused_port(hostname)
+class NetworksConfigConstructor:
+    def __init__(self, file_path=None):
+        self.networks = {}
+        self.used_sockets = []
+        self.file_path = file_path
+        if self.file_path is not None:
+            self.read_from_file()
+
+    def add_node(self, node_name, network_name="default", app_hostname=None, cqc_hostname=None, vnode_hostname=None,
```

Comment:
You removed the documentation here. Is it because all parameters drastically changed?

Reply:
Yes, I still need to add quite a few doc-strings